test/boost/memtable_test.cc

```diff
@@ -572,6 +573,74 @@ SEASTAR_THREAD_TEST_CASE(test_tombstone_compaction_during_flush) {
     mt->cleaner().drain().get();
 }
 
+SEASTAR_THREAD_TEST_CASE(test_range_tombstones_are_compacted_with_data) {
+    tests::reader_concurrency_semaphore_wrapper semaphore;
+    simple_schema ss;
+    auto s = ss.schema();
+    auto mt = make_lw_shared<replica::memtable>(ss.schema());
+
+    auto pk = ss.make_pkey(0);
+    auto pr = dht::partition_range::make_singular(pk);
+
+    auto old_tombstone = ss.new_tombstone(); // older than any write, does not cover anything
+
+    {
+        mutation m(ss.schema(), pk);
+        ss.add_row(m, ss.make_ckey(1), "v1");
+        ss.add_row(m, ss.make_ckey(2), "v1");
+        ss.add_row(m, ss.make_ckey(3), "v1");
+        ss.add_row(m, ss.make_ckey(4), "v1");
+        mt->apply(m);
```

Comment:
Shouldn't it be much larger to force preemption?

Reply:
Not in debug mode.